### PR TITLE
Add a glossary entry on colorization.

### LINKED ISSUE #6655: Explain what "colorization" means in namespace GridGenerator
That's a term that I don't find obvious. We should just have a glossary entry for it and link to it each time we use that word.

### PR DESCRIPTION
**Pull request by drwells**
Fixes #6655.

### CODE REVIEW

File: doc/doxygen/headers/glossary.h [hunk 1]
```diff
@@ -378,6 +380,28 @@
...
+ * <ol>
+ *   <li> Most of the functions in the GridGenerator namespace take an optional
+ *   argument <code>colorize</code>. This argument controls whether or not the
+ *   different parts of the boundary will be assigned different
+ *   @ref GlossBoundaryIndicator "boundary indicators". Some functions also assign
+ *   different
+ *   @ref GlossMaterialId "material indicators" as well.</li>
+ *   <li> The function GraphColoring::make_graph_coloring computes a
+ *   decomposition of a Triangulation (more exactly, a range of iterators). No
+ *   two adjacent cells are given the same color.
```
Comment (jppelteret) [inline doc/doxygen/headers/glossary.h]: Missing an `</li>`? (There's one three lines above.)

File: include/deal.II/grid/grid_generator.h [hunk 2]
```diff
@@ -466,7 +479,9 @@ namespace GridGenerator
    * Each entry must be positive. An empty vector is equivalent to one
    * subdivision in each direction.
    *
-   * @param colorize Assign different boundary ids if set to true.
+   * @param colorize
+   * (see @ref GlossColorization "the glossary entry on colorization")
```
Comment (jppelteret) [inline include/deal.II/grid/grid_generator.h]: Maybe this phrase in parentheses should go after the definition of the parameter (you've done this for `hyper_cube_with_cylindrical_hole`)

**Comment by drwells** [comment]
Thanks @jppelteret; those all need to be fixed. I'll post here when this is done.

**Comment by drwells** [comment]
I fixed the missing `</li>` and also moved the citation to the end of each sentence.

File: doc/doxygen/headers/glossary.h [hunk 3]
```diff
@@ -378,6 +380,28 @@
...
+ * coloring:
+ *
+ * <ol>
+ *   <li> Most of the functions in the GridGenerator namespace take an optional
+ *   argument <code>colorize</code>. This argument controls whether or not the
+ *   different parts of the boundary will be assigned different
+ *   @ref GlossBoundaryIndicator "boundary indicators". Some functions also assign
+ *   different
+ *   @ref GlossMaterialId "material indicators" as well.</li>
+ *   <li> The function GraphColoring::make_graph_coloring computes a
```
Comment (jppelteret) [inline doc/doxygen/headers/glossary.h]: I think that `GraphColoring::make_graph_coloring` should be `GraphColoring::make_graph_coloring()` for doxygen to pick it up.

Comment (drwells) [reply doc/doxygen/headers/glossary.h]: Thats easy enough to fix; done!

**Comment by drwells** [comment]
Per @jppelteret remarks I will mark this as reviewed and ready to merge.

File: doc/doxygen/headers/glossary.h [hunk 4]
```diff
@@ -378,6 +380,28 @@
  * @see @ref boundary "The module on boundaries"
  *
  *
+ * <dt class="glossary">@anchor GlossColorization <b>Colorization</b></dt>
+ * <dd><em>Colorization</em> is the process of marking certain parts of a
+ * Triangulation with different labels. The use of the word <em>color</em>
+ * comes from cartography, where countries on a map are made visually distinct
+ * from each other by assigning them different colors. Using the same term
+ * <em>coloring</em> is common in mathematics, even though we assign integers
+ * and not hues to different regions. deal.II refers to two processes as
```
Comment (bangerth) [inline doc/doxygen/headers/glossary.h]: I love this explanation! I never thought about it in these terms -- thank you for making that leap for me!

**Pull request merged**